examples/plugins/traffic-control/docker.go

```diff
@@ -0,0 +1,144 @@
+package main
+
+import (
+	"fmt"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	docker "github.com/fsouza/go-dockerclient"
+)
+
+// DockerClient internal data structure
+type DockerClient struct {
+	store  *Store
+	client *docker.Client
+}
+
+// NewDockerClient instantiates a new DockerClient
+func NewDockerClient(store *Store) (*DockerClient, error) {
+	dc, err := docker.NewClient("unix:///var/run/docker.sock")
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to docker daemon: %v", err)
+	}
+	return &DockerClient{
+		store:  store,
+		client: dc,
+	}, nil
+}
+
+// Start docker client
+func (c *DockerClient) Start() {
+	for {
+		func() {
```

Comment:
Ah yeah, no you are right. My mistake.

On Fri, Sep 9, 2016 at 2:47 PM Alessandro Puccetti <notifications@github.com>
wrote:

> In examples/plugins/traffic-control/docker.go
> <https://github.com/weaveworks/scope/pull/1837#discussion_r78182169>:
>
> > +// NewDockerClient instantiates a new DockerClient
> > +func NewDockerClient(store *Store) (*DockerClient, error) {
> > +	dc, err := docker.NewClient("unix:///var/run/docker.sock")
> > +	if err != nil {
> > +		return nil, fmt.Errorf("failed to connect to docker daemon: %v", err)
> > +	}
> > +	return &DockerClient{
> > +		store:  store,
> > +		client: dc,
> > +	}, nil
> > +}
> > +
> > +// Start docker client
> > +func (c *DockerClient) Start() {
> > +	for {
> > +		func() {
>
> No, because `continue` will start over from the top of the for loop and
> the cleanup would not be executed. Every return needs to trigger the clean
> up because otherwise there are cases where `events` are not cleaned up.
> This is also the reason why I did not see the advantage of putting the
> function inline.
> … <#m_2828333680483081382_>
> On Fri, Sep 9, 2016 at 3:18 PM, Paul Bellamy ***@***.***> wrote: In
> examples/plugins/traffic-control/docker.go <#1837 (comment)
> <https://github.com/weaveworks/scope/pull/1837#discussion_r78177891>>: >
> +// NewDockerClient instantiates a new DockerClient > +func
> NewDockerClient(store *Store) (*DockerClient, error) { > + dc, err :=
> docker.NewClient("unix:///var/run/docker.sock") > + if err != nil { > +
> return nil, fmt.Errorf("failed to connect to docker daemon: %v", err) > + }
> > + return &DockerClient{ > + store: store, > + client: dc, > + }, nil > +}
> > + > +// Start docker client > +func (c *DockerClient) Start() { > + for {
> > + func() { If you replace the returns with continues, and remove the
> defer (by moving the cleanup to after the for loop), we wouldn't need the
> lambda here. — You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <
> https://github.com/weaveworks/scope/pull/1837/files/f2c2ce9d33416d849c9b6ec36600e7cdbd035015#r78177891>,
> or mute the thread <
> https://github.com/notifications/unsubscribe-auth/AGRc74emazOAbjn8Xfa6Unugw9qJbsnJks5qoVy_gaJpZM4JyDW_>
> .
> -- Alessandro Puccetti
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/weaveworks/scope/pull/1837/files/f2c2ce9d33416d849c9b6ec36600e7cdbd035015#r78182169>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAPRVzIPQuT_jY0tA3hgNLwYPIbIP_XYks5qoWN3gaJpZM4JyDW_>
> .
>